File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {

    int size = clothes.size();
    unordered_map <string, int> map; // 의상 종류, 개수
    
    for(auto cloth: clothes){
        map[cloth[1]] ++;
    }
    int answer = 1;

    for(auto pair: map)
        answer *= (pair.second+1);
    
    
    return answer - 1;
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    dict = {}
    for item,type in clothes:
        if type in dict:
            dict[type] += 1
        else:
            dict[type] = 1
    
    answer = 1
    
    for _, value in dict.items():
        answer *= (value + 1)
    
    return answer - 1